mesonbuild/utils/universal.py

```diff
@@ -1283,7 +1302,7 @@ def line_is_valid(line: str, variable_format: str) -> bool:
     # during substitution so we can warn the user to use the `copy:` kwarg.
     confdata_useless = not confdata.keys()
     for line in data:
-        if line.startswith(search_token):
+        if line.lstrip().startswith(search_token):
```

Comment:
This is for supporting indented cmakedefines.
Its not officially mentioned in the docs but leading spaces between the start of the line and the `#` are allowed and mimick the behavior of `#define`.
The note for spaces between `#` and `cmakdefine` exists because of an oversight prior to 3.10 that did not allow it, despite `#define` doing do.